sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/BlobHelper.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Azure.Containers.ContainerRegistry.Specialized
+{
+    internal class BlobHelper
+    {
+        internal static string ComputeDigest(Stream stream)
+        {
+            // According to https://docs.docker.com/registry/spec/api/#content-digests, compliant
+            // registry implementations use sha256.
+
+            using SHA256 sha256 = SHA256.Create();
+            var position = stream.Position;
+
+            try
+            {
+                stream.Position = 0;
+                var hashValue = sha256.ComputeHash(stream);
+                return FormatDigest(hashValue);
+            }
+            finally
+            {
+                stream.Position = position;
+            }
+        }
+
+        internal static string FormatDigest(byte[] hash)
+        {
+            return BytesToString(hash);
+        }
+
+        private static string BytesToString(byte[] bytes)
```

Comment:
this routine is pretty slow in comparison to what is possible if we write our own formatting routine. Do we care about perf here?

Reply:
I'm not sure this is our primary bottleneck, but tracking this with https://github.com/Azure/azure-sdk-for-net/issues/33614 and we can prioritize per customer needs.  Thanks @KrzysztofCwalina!